src/solvers/td3.ml

```diff
@@ -272,6 +308,9 @@ module WP =
         let tmp = simple_solve l x y in
         if HM.mem rho y then add_infl y x;
         if tracing then trace "sol2" "eval %a ## %a -> %a\n" S.Var.pretty_trace x S.Var.pretty_trace y S.Dom.pretty tmp;
+        if skip_unchanged_rhs then
+          (let (oldv,curr_dep_vals) = HM.find dep_vals x in
+           HM.replace dep_vals x (oldv,((y,tmp) :: curr_dep_vals)));
```

Comment:
I suspect this association list structure could also be very inefficient and memory hungry. If the right-hand side `get`s a single unknown 1000 times and the value doesn't change in between, then the resulting list has 1000 identical pairs.
Maybe this is what is happening with some of the slower and more complicated privatizations since they rely on a lot more global invariants. For normal evaluation it isn't a problem because without changes, the subsequent `get`s of the same variable are direct lookups of a stable variable, so they're as efficient as they possibly could be (the solver itself serves as a cache for `get`).